File: tp1/aux_functions/c/FIM_flag.c
#include <stdio.h>
#include <stdbool.h>
#define BUFFER 128

	bool stringComp(char *str, char *flag, int len);

	int main(){
	
		char flag[] = "FIM";
		char str[BUFFER];
		
		int len = ((sizeof(flag) / sizeof(flag[0])) - 1);

		while (!stringComp(str, flag, len)){
			fgets(str, BUFFER, stdin);
		}

	}

	bool stringComp(char *str, char *flag, int len){

		if (str[len] != '\n' && str[len] != '\0')
			return false;
		
		for (int i = 0; i < len; i++){
			if (str[i] != flag[i])
				return false;
		}

		return true;
	}

File: tp1/aux_functions/java/FIM_flag.java
import java.util.Scanner;

public class FIM_flag {

	public static void main(String[] args){

	String flag = "FIM";
	String str;

	Scanner scanner = new Scanner(System.in);

	do {
		str = scanner.nextLine();
	} while (!stringComp(str, flag));

}


	static boolean stringComp(String str, String flag){

		if (str.length() != flag.length())
			return false;

		for (int i = 0; i < flag.length(); i++){
			if (str.charAt(i) != flag.charAt(i))
				return false;
		}

		return true;
	}

}
